src/server/models/Reading.js

```diff
@@ -54,6 +55,34 @@ class Reading {
 		return getDB().none(sqlFile('reading/create_function_get_barchart_readings.sql'));
 	}
 
+	/**
+	 * Returns a promise to create the function and materialized views that aggregate
+	 * readings by various time intervals.
+	 * @return {Promise<void>}
+	 */
+	static createCompressedReadingsMaterializedViews(conn = getDB) {
```

Comment:
This still isn't going to work, as I've mentioned in other reviews and possibly an issue a while back. The argument here needs to be compatible with the argument you get in the callback from pg-promise's `db.tx`, otherwise there's no way to wrap the call in a transaction (unless you wrap the transaction object, but that seems like a bad solution). If you want to inject the db in this way you might be able to get away with wrapping the whole class declaration in a function that gets passed db and then calling it with (getDB()) and exporting the result. But I feel like you should explore mocking it as it gets exported instead.

Reply:
So, the basic issue here is that all these functions want a default connection. Perhaps simply striking that out and forcing any usage of anything that touches the database to pass in a connection would work better?

This actually has a few advantages. First of all, it would definitely remove the use of getDB. Second, it would solve the problem of race conditions for database connection initialization, which was why we moved to getDB in the first place. Third, it would make explicit any use of the database, which is almost certainly a good idea anyway.